docs/skill-guide.md

```diff
@@ -0,0 +1,230 @@
+The <a href="../scaffolding/">scaffolding tool</a> allows you create the folder structure required for a skill.
+
+!!!	Note
+	Before developing your first skill, please read the <a href="../skill/">skill guide</a>.
+
+## Step 1: Setup
+
+Ensure, you have followed the <a href="../quickstart/">preliminaries</a>. We will first create an agent and add a scaffold skill, which we call `my_search`:
+
+``` bash
+aea create my_agent && cd my_agent
+aea scaffold skill my_search
+```
+
+## Step 2: Develop a Behaviour
+
+A `Behaviour` class contains the business logic specific to initial actions initiated by the agent rather than reactions to other events.
+
+In this example, we implement a simple search behaviour. Each time, `act()` gets called, we will send a search request to the OEF.
+
+``` python
+from aea.protocols.oef.message import OEFMessage
+from aea.protocols.oef.models import Query, Constraint, ConstraintType
+from aea.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from aea.skills.base import Behaviour
+
+
+class MySearchBehaviour(Behaviour):
+    """This class provides a simple search behaviour."""
+
+    def __init__(self, **kwargs):
+        """Initialize the search behaviour."""
+        super().__init__(**kwargs)
+        self.sent_search_count = 0
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+        
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+        self.sent_search_count += 1
+        search_constraints = [Constraint("country",
+        	                  ConstraintType("==", "UK"))]
+        search_query_w_empty_model = Query(search_constraints, model=None)
+        search_request = OEFMessage(oef_type=OEFMessage.Type.SEARCH_SERVICES,
+                                    id=self.sent_search_count,
+                                    query=search_query_w_empty_model)
+        self.context.outbox.put_message(to=DEFAULT_OEF,
+                                        sender=self.context.agent_address,
```

Comment:
this line gives me:
```
AttributeError: 'SkillContext' object has no attribute 'agent_address'
```

Reply:
This is because we haven't merged @Totoual 's branch yet